modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/CurrencySnapshotCreator.scala

```diff
@@ -50,137 +54,181 @@ trait CurrencySnapshotCreator[F[_]] {
     lastContext: CurrencySnapshotContext,
     trigger: ConsensusTrigger,
     events: Set[CurrencySnapshotEvent],
-    rewards: Option[Rewards[F, CurrencySnapshotStateProof, CurrencyIncrementalSnapshot, CurrencySnapshotEvent]]
-  ): F[CurrencySnapshotCreationResult]
+    rewards: Option[Rewards[F, CurrencySnapshotStateProof, CurrencyIncrementalSnapshot, CurrencySnapshotEvent]],
+    facilitators: Set[PeerId]
+  ): F[CurrencySnapshotCreationResult[CurrencySnapshotEvent]]
 }
 
 object CurrencySnapshotCreator {
   def make[F[_]: Async: KryoSerializer: SecurityProvider](
     currencySnapshotAcceptanceManager: CurrencySnapshotAcceptanceManager[F],
-    maybeDataApplicationService: Option[(L0NodeContext[F], BaseDataApplicationL0Service[F])]
+    maybeDataApplicationService: Option[(L0NodeContext[F], BaseDataApplicationL0Service[F])],
+    snapshotSizeConfig: SnapshotSizeConfig
   ): CurrencySnapshotCreator[F] = new CurrencySnapshotCreator[F] {
     private val logger = Slf4jLogger.getLogger
 
+    private def maxProposalSizeInBytes(facilitators: Set[PeerId]) =
```

Comment:
What's the return type? Will this function return value < 0?

Reply:
Max proposal size can't be `<=  0` so I would set it to `PosLong`.